janus-idp/backstage-plugins
# feat: add custom processor module to add ScaffoldedFrom relation

### Description
Adds a custom catalog processor that creates relations between catalog entities and the scaffolder templates that created them based on the `spec.scaffoldedFrom` field in the `catalog-info.yaml` of the catalog entities.

![image](https://github.com/janus-idp/backstage-showcase/assets/50030060/ff469405-6bd6-4452-8468-eecc9a873d2c)

### What issue(s) does this fix?
Fixes: [RHIDP-2156](https://issues.redhat.com/browse/RHIDP-2156)

## How to test changes / Special notes to the reviewer

1. Install the module using the instructions in the README.md
2. Ingest an entity with a `spec.scaffoldedFrom` field that matches the `entityRef` of a template entity in the catalog (ex: `template:default/test-template`). You can potentially use the following example template: https://github.com/janus-idp/backstage-plugins/blob/main/plugins/scaffolder-annotator-action/examples/templates/01-scaffolder-template.yaml
3. Observe the relation graph of the entity in the catalog (or view the raw YAML/JSON of the entity and find the `scaffoldedFrom` relation in the `relations` array)
